nbconvert/exporters/templateexporter.py

```diff
@@ -250,10 +272,13 @@ def __init__(self, config=None, **kw):
 
     def _load_template(self):
         """Load the Jinja template object from the template file
-        
+
         This is triggered by various trait changes that would change the template.
         """
 
+        if self.raw_template:
+            self._register_raw_template(self.raw_template)
```

Comment:
I'm a bit uncomfortable that this is called both in `_load_template()` and in the observe handler. I haven't identified a failure case yet, but I suspect that this is going to make it harder to reason about what data is where at what point in execution.

In particular, I think that using a traitlet but encouraging people to override it in subclasses with a simple attribute is likely to be a source of confusion - I think the mechanism should either use traitlets entirely, or not involve a traitlet at all. (The latter is my preference, but that's because I generally think traitlets introduce more confusion than they remove)